### PR TITLE
Activate .NET code analysis

### PR DESCRIPTION
**Pull request by Regenhardt**
Taken from the open BaGet PR https://github.com/loic-sharma/BaGet/pull/784

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -21,6 +21,12 @@
     <NoWarn>$(NoWarn);1591</NoWarn>
   </PropertyGroup>
 
+  <!-- .NET source code analysis https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview -->
+  <PropertyGroup>
```
Comment (FroggieFrog) [inline src/Directory.Build.props:25]: As mentioned in the above link, this should be unnecessary.
> If your project targets .NET 5 or later, code analysis is enabled by default.

Comment (seriouz) [reply src/Directory.Build.props:25]: May these lines trigger the GitHub Actions Check?
I've never seen these GitHub check annotations on this repo.

Comment (Regenhardt) [reply src/Directory.Build.props:25]: The build/test one? That one is defined in https://github.com/bagetter/BaGetter/blob/main/.github/workflows/main.yml

Comment (Regenhardt) [reply src/Directory.Build.props:25]: Enabling is not necessary, but setting the level to minimum is still a good idea so I kept that one.

**Comment by FroggieFrog** [comment]
According to the build workflow, there is still one issue left:
grafik

**Comment by FroggieFrog** [comment]
Maybe you can integrate 2 more changes from PR #46 (then #46 is superfluous and can be closed)
1. in `src/BaGetter.Core/Entities/Converters/StringArrayToJsonConverter.cs`: line 14 `v => (!string.IsNullOrEmpty(v)) ? JsonConvert.DeserializeObject<string[]>(v) : Array.Empty<string>())`
2. in `tests/BaGetter.Tests/HostIntegrationTests.cs`: line 65 `.CreateHostBuilder(Array.Empty<string>())`

**Review by FroggieFrog — APPROVED** [review]

**Comment by seriouz** [comment]
@Regenhardt Can you merge?

**Pull request merged**